python/lsst/daf/butler/core/dimensions/graph.py

```diff
@@ -0,0 +1,473 @@
+
+    def __le__(self, other):
+        return self._dimensions <= other
+
+    def __lt__(self, other):
+        return self._dimensions < other
+
+    def __ge__(self, other):
+        return self._dimensions >= other
+
+    def __gt__(self, other):
+        return self._dimensions > other
+
+    def issubset(self, other):
+        """Return `True` if all dimensions in ``self`` are also in ``other``.
+
+        The empty graph is a subset of all graphs (including the empty graph).
+        """
+        return self <= other
+
+    def issuperset(self, other):
+        """Return `True` if all dimensions in ``other`` are also in ``self``,
+        and `False` otherwise.
+
+        All graphs (including the empty graph) are supersets of the empty
+        graph.
+        """
+        return self >= other
+
+    def isdisjoint(self, other):
+        """Return `True` if there are no dimensions in both ``self`` and
+        ``other``, and `False` otherwise.
+
+        All graphs (including the empty graph) are disjoint with the empty
+        graph.
+        """
+        return self._dimensions.isdisjoint(other)
+
+    def union(self, *others, optional=False):
+        """Return a new graph containing all elements that are in ``self`` or
+        any of the other given graphs.
+
+        Parameters
+        ----------
+        *others : iterable over `Dimension` or `str`.
+            Other sets whose elements should be included in the result.
+
+        Returns
+        -------
+        result : `DimensionGraph`
+            Graph containing any elements in any of the inputs.
+        """
+        return DimensionGraph(self.universe, self._dimensions.union(*others), optional=optional)
+
+    def intersection(self, *others, optional=False):
+        """Return a new graph containing all elements that are in both  ``self``
+        and all of the other given graphs.
+
+        Parameters
+        ----------
+       *others : iterable over `Dimension` or `str`.
+            Other sets whose elements may be included in the result.
+
+        Returns
+        -------
+        result : `DimensionGraph`
+            Graph containing all elements in all of the inputs.
+        """
+        return DimensionGraph(self.universe, self._dimensions.intersection(*others), optional=optional)
+
+    def __or__(self, other):
+        if isinstance(other, DimensionGraph):
+            return self.union(other)
+        return NotImplemented
+
+    def __and__(self, other):
+        if isinstance(other, DimensionGraph):
+            return self.intersection(other)
+        return NotImplemented
+
+    def extract(self, elements, optional=False):
+        r"""Return a new graph containing the given elements.
+
+        Parameters
+        ----------
+        elements : iterable of `DimensionElement` or `str`
+            `Dimension`\s, `DimensionJoin`\s, or names thereof.  These must be
+            in ``self`` or ``self.joins()``.
```

Comment:
There is a also support for `DimensionGraph` in the code 🙂

Reply:
Yes, but `DimensionGraph` *is* an iterable of `DimensionElement`.